spec/acceptance/datasets_spec.rb

```diff
@@ -0,0 +1,215 @@
+end
+
+# https://github.com/zipmark/rspec_api_documentation
+resource 'Datasets' do
+
+  header 'Accept', 'application/json'
+  header 'Content-Type', 'application/json'
+  header 'Authorization', :authentication_token
+
+  let(:format) { 'json' }
+
+  create_entire_hierarchy
+
+  # Create post parameters from factory
+  let(:post_attributes) { FactoryGirl.attributes_for(:dataset, name: 'New Dataset name') }
+
+  ################################
+  # INDEX
+  ################################
+
+  # Expected count is 2 because of the 'default dataset', which is seed data added during a clean install.
+
+  get '/datasets' do
+    let(:authentication_token) { admin_token }
+    standard_request_options(:get, 'INDEX (as admin)', :ok, {expected_json_path: 'data/0/name', data_item_count: 2})
+  end
+
+  # will only return dataset that are created by the user
+
+  get '/datasets' do
+    let(:authentication_token) { reader_token }
+    standard_request_options(:get, 'INDEX (as non admin user)', :ok, {response_body_content: '200', data_item_count: 0})
+  end
+
+  get '/datasets' do
+    let(:authentication_token) { invalid_token }
+    standard_request_options(:get, 'INDEX (with invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  get '/datasets' do
+    standard_request_options(:get, 'INDEX (as anonymous user)', :ok, {remove_auth: true, response_body_content: '200', data_item_count: 0})
+  end
+
+  ################################
+  # CREATE
+  ################################
+
+  post '/datasets' do
+    body_params
+    let(:raw_post) { {'dataset' => post_attributes}.to_json }
+    let(:authentication_token) { admin_token }
+    standard_request_options(:post, 'CREATE (as admin)', :created, {expected_json_path: 'data/name', response_body_content: 'New Dataset name'})
+  end
+
+  post '/datasets' do
+    body_params
+    let(:raw_post) { {'dataset' => post_attributes}.to_json }
+    let(:authentication_token) { reader_token }
+    standard_request_options(:post, 'CREATE (as non admin user)', :created, {expected_json_path: 'data/name', response_body_content: 'New Dataset name'})
+  end
+
+  post '/datasets' do
+    body_params
+    let(:raw_post) { {'dataset' => post_attributes}.to_json }
+    let(:authentication_token) { invalid_token }
+    standard_request_options(:post, 'CREATE (invalid token)', :unauthorized, {expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  post '/datasets' do
+    body_params
+    let(:raw_post) { {'dataset' => post_attributes}.to_json }
+    standard_request_options(:post, 'CREATE (as anonymous user)', :unauthorized, {remove_auth: true, expected_json_path: get_json_error_path(:sign_up)})
+  end
+
+  ################################
+  # NEW
+  ################################
+
+  get '/datasets/new' do
+    let(:authentication_token) { admin_token }
+    standard_request_options(:get, 'NEW (as admin)', :ok, {expected_json_path: 'data/name/'})
```

Comment:
as with `dataset_items` new should be accessible by all  users

Reply:
#new is now available to all users including guest